tests/render/renderer/test_notebook_renderer.py

```diff
@@ -1495,3 +1500,76 @@ def test_complex_suite(warning_suite):
     for obs_cell, expected_cell in zip(obs["cells"], expected["cells"]):
         assert obs_cell == expected_cell
     assert obs == expected
+
+
+def test_notebook_execution_with_pandas_backend(titanic_data_context):
+    """
+    To set this test up we:
+
+    - create a suite
+    - add a few expectations (both table and column level)
+    - verify that no validations have happened
+    - create the suite edit notebook by hijacking the private cli method
+
+
+    We then:
+    - execute that notebook (Note this will raise various errors like
+    CellExecutionError if any cell in the notebook fails
+    - create a new context from disk
+    - a validation has been run with our expectation suite
+    """
+    context = titanic_data_context
+    root_dir = context.root_directory
+    uncommitted_dir = os.path.join(root_dir, "uncommitted")
+    suite_name = "warning"
+
+    context.create_expectation_suite(suite_name)
+    csv_path = os.path.join(root_dir, "..", "data", "Titanic.csv")
+    batch_kwargs = {"datasource": "mydatasource", "path": csv_path}
+    batch = context.get_batch(batch_kwargs, suite_name)
+    batch.expect_table_column_count_to_equal(1)
+    batch.expect_table_row_count_to_equal(1313)
+    batch.expect_column_values_to_be_in_set("Sex", ["female", "male"])
+    batch.save_expectation_suite(discard_failed_expectations=False)
```

Comment:
Could we also save the expectation suite here and the immediately delete from disk after calling the edit suite command to ensure that the notebook actually creates the new suite by being run?

Reply:
I wish we could, but the notebook itself reads the suite from disk. I thought about cracking the notebook open but it seemed like too much work and could introduce other errors.